### PR TITLE
PMM-6969 rules api model

### PR DESCRIPTION
**Pull request by palash25**
[PMM-XXXX](https://jira.percona.com/browse/PMM-XXXX)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/0

- [ ] Links to other linked pull requests (optional).
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

**Comment by palash25** [comment]
do not review yet, build fails

File: models/rule.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package model
```
Comment (AlekSi) [inline models/rule.go:17]:
```suggestion
package models
```

File: models/database.go [hunk 2]
```diff
@@ -346,6 +346,23 @@ var databaseSchema = [][]string{
...
+			template JSONB,
+			summary VARCHAR,
+			disabled BOOLEAN,
+			params JSONB,
+			for VARCHAR NOT NULL,
+			severity VARCHAR,
+			custom_labels JSONB,
+			filters JSONB,
+			channels JSONB,
+			created_at VARCHAR NOT NULL,
```
Comment (palash25) [inline models/database.go:361]: todo: change to timestamp

File: models/rule.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	iav1beta1 "github.com/percona/pmm/api/managementpb/ia"
+	"gopkg.in/reform.v1"
+)
+
+//go:generate reform
+
+// Rule represents alert rule configuration.
+//reform:ia_rules
+type Rule struct {
+	Template     *iav1beta1.Template    `reform:"template"`
```
Comment (AlekSi) [inline models/rule.go:33]: Don't use API types in database models – they can't be serialised correctly. That's why reform explodes

File: models/database.go [hunk 4]
```diff
@@ -371,6 +371,25 @@ var databaseSchema = [][]string{
...
+	23: {
+		`CREATE TABLE ia_rules (
+			id VARCHAR NOT NULL,
+			template JSONB,
+			summary VARCHAR,
+			disabled BOOLEAN,
+			params JSONB,
+			"for" INTERVAL NOT NULL,
+			severity VARCHAR,
+			custom_labels JSONB,
```
Comment (palash25) [inline models/database.go:384]: add annotations

File: models/rule_helpers.go [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+		Template: Template{
+			Name: params.TemplateName,
+		},
+		Disabled: params.Disabled,
+		For:      params.For.AsDuration(),
+		Severity: params.Severity.String(),
+		Filters:  params.Filters,
+	}
+
+	labels, err := json.Marshal(params.CustomLabels)
```
Comment (palash25) [inline models/rule_helpers.go:113]: TODO: use setLabels and setAnnotations functions like the template api does

File: models/rule_helpers.go [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+	err := q.Delete(&Rule{ID: id})
+	if err != nil {
+		return errors.Wrap(err, "failed to delete alert Rule")
+	}
+	return nil
+}
+
+// TODO remove once template API is merged
+func makeModelTemplate(params *CreateRuleParams) *Template {
```
Comment (palash25) [inline models/rule_helpers.go:201]: writing a dummy template to the DB for now, will use template model helpers once 6909 is merged

**Comment by artemgavrilov** [comment]
https://github.com/percona/pmm-managed/pull/554

**Pull request closed without merging**